internal/controllers/synthesis/gc.go

```diff
@@ -135,6 +135,18 @@ func (p *podGarbageCollector) deletePod(ctx context.Context, pod *corev1.Pod, lo
 	if len(pod.Status.ContainerStatuses) > 0 {
 		logger = logger.WithValues("restarts", pod.Status.ContainerStatuses[0].RestartCount)
 	}
+
+	// Check for image pull errors before deleting the pod
+	if errMsg, hasError := checkImagePullError(pod); hasError {
```

Comment:
We should set the error status as soon as the container enters the image pull state. Don't wait until deletion time.